snuba/utils/streams/consumers/consumer.py

```diff
@@ -91,6 +91,22 @@ def seek(self, offsets: Mapping[TStream, TOffset]) -> None:
         """
         return self.__backend.seek(offsets)
 
+    def pause(self, streams: Sequence[TStream]) -> None:
+        """
+        Pause the consumption of messages for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        return self.__backend.pause(streams)
+
+    def resume(self, streams: Sequence[TStream]) -> None:
+        """
+        Resume the consumption of messages for the provided streams.
+
+        Raises a ``RuntimeError`` if called on a closed consumer.
+        """
+        return self.__backend.resume(streams)
+
```

Comment:
> That feature would have been useful, have you considered supporting a concept of `pause` `resume` for the batching consumer ?

Not really. I guess it would be possible to implement via some sort of sentinel value or exception thrown by `AbstractBatchWorker.process_message`, but it seems like the more control the worker has over the consumer behavior, the less point there is to using inversion of control, no?